src/ranged.cpp

```diff
@@ -112,17 +112,15 @@ size_t blood_trail_len( int damage )
 }
 
 dealt_projectile_attack Creature::projectile_attack( const projectile &proj_arg, const tripoint &source,
-                                                     const tripoint &target_arg, double shot_dispersion )
+                                                     const tripoint &target_arg, double dispersion )
 {
     const bool do_animation = OPTIONS["ANIMATIONS"];
 
-    // shot_dispersion is in MOA - 1/60 of a degree
-    constexpr double moa = M_PI / 180 / 60;
-    // We're approximating the tangent. Multiplying angle*range by ~0.745 does that (kinda).
-    constexpr double to_tangent = 0.745;
-
+    // for the isosceles triangle formed by the intended and actual targets
+    // we can use the cosine formula (a² = b² + c² - 2bc⋅cosθ) to calculate the tangent
     double range = rl_dist( source, target_arg );
-    double missed_by = shot_dispersion * moa * to_tangent * range;
+    double missed_by = sqrt( 2 * pow( range, 2 ) - ( 2 * pow( range, 2 ) * cos( ARCMIN( dispersion ) ) ) );
```

Comment:
Can be simplified to:
```
sqrt( 2 * ( range * range ) * ( 1 - cos( ARCMIN( dispersion ) ) );
```

The formula is ~10% less forgiving than the current one. That's quite important but not a huge deal.